api/src/plugins/email-token-scheme.js

```diff
@@ -7,9 +7,12 @@ exports.register = function (server, pluginOptions, next) {
       async authenticate(request, reply) {
         try {
           var token = await schemeOptions.findToken(request.query.token);
+          if (token.name === 'TypeError') {
```

Comment:
there's already a `try {} catch()` for error handling. The `findToken` function should throw its error and the catch block of this function can handle the error however you want.

It looks like this is catching an error caused by a genuine bug or fault in the code and disguising it as a 400? I think the best way to deal with this isn't to try and hide this specific error but to prevent it from happening in the first place with proper checks in the `findToken` function to prevent this TypeError from occurring in the first place